fiberplane-models/src/providers/schema/fields/array_field.rs

```diff
@@ -0,0 +1,140 @@
+    /// Leaving the minimum_length to 0 makes the whole field optional.
+    #[builder(default = 0)]
+    pub minimum_length: u32,
+    /// The maximum number of entries the array can have and still be valid.
+    ///
+    /// It is None when there is no maximum number
+    #[builder(default, setter(strip_option))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub maximum_length: Option<u32>,
+    /// The schema of the elements inside a row of the array.
+    ///
+    /// ### Accessing row fields
+    ///
+    /// The name of each QueryField inside the element_schema can be used as
+    /// an indexing key for a field. That means that if `element_schema` contains
+    /// a [TextField](crate::providers::TextField) with the name `parameter_name`,
+    /// then you will be able to access the value of that field using
+    /// `ArrayField::get(i)::get("parameter_name")` for the i-th element.
+    ///
+    /// ### Serialization
+    ///
+    /// For example if an array field has this `element_schema`:
+    /// ```rust,no_run
+    /// # use fiberplane_models::providers::{ArrayField, TextField, SelectField, IntegerField};
+    /// ArrayField::builder()
+    ///   .name("table".to_string())
+    ///   .label("example".to_string())
+    ///   .element_schema(vec![
+    ///     TextField::new().with_name("key").into(),
+    ///     SelectField::new().with_name("operator").with_options(&["<", ">", "<=", ">=", "=="]).into(),
+    ///     IntegerField::new().with_name("value").into(),
+    ///   ]).build();
+    /// ```
+    ///
+    /// Then the URL-encoded serialization for the fields is expected to look like this
+    /// (line breaks are only kept for legibility):
+    /// ```txt
+    ///  "table[0][key]=less+than&
+    ///  table[2][operator]=%3E&
+    ///  table[0][operator]=%3C&
+    ///  table[2][key]=greater+than&
+    ///  table[2][value]=10&
+    ///  table[0][value]=12"
+    /// ```
+    ///
+    /// Note that we are allowed to skip indices, this serialization is expected to
+    /// be read as:
+    /// ```rust,no_run
+    /// # #[derive(Debug, PartialEq)]
+    /// # struct Row { key: String, operator: String, value: u32 }
+    /// # let table: Vec<Row> = vec![];
+    /// assert_eq!(table, vec![
+    ///   Row {
+    ///     key: "less than".to_string(),
+    ///     operator: "<".to_string(),
+    ///     value: 12,
+    ///   },
+    ///   Row {
+    ///     key: "greater than".to_string(),
+    ///     operator: ">".to_string(),
+    ///     value: 10,
+    ///   },
+    /// ]);
+    /// ```
+    ///
+    /// ### Required row fields
+    ///
+    /// Any field that is marked as `required` inside `element_schema` makes it
+    /// mandatory to create a valid row to the Array Field.
+    pub element_schema: QuerySchema,
+}
+
+impl ArrayField {
+    /// Creates a new array field with all default values.
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn with_label(self, label: &str) -> Self {
```

Comment:
```suggestion
    pub fn with_label(self, label: impl Into<String>) -> Self {
```

I will also update the other field types to make sure their methods use this consistently as well.

(Same goes for the other methods accepting strings)